src/rgw/rgw_acl.h

```diff
@@ -327,35 +311,18 @@ class ACLOwner
   void dump(Formatter *f) const;
   void decode_json(JSONObj *obj);
   static void generate_test_instances(std::list<ACLOwner*>& o);
-  void set_id(const rgw_user& _id) { id = _id; }
-  void set_name(const std::string& name) { display_name = name; }
-
-  rgw_user& get_id() { return id; }
-  const rgw_user& get_id() const { return id; }
-  std::string& get_display_name() { return display_name; }
-  const std::string& get_display_name() const { return display_name; }
-  friend bool operator==(const ACLOwner& lhs, const ACLOwner& rhs);
-  friend bool operator!=(const ACLOwner& lhs, const ACLOwner& rhs);
+
+  auto operator<=>(const ACLOwner&) const = default;
 };
 WRITE_CLASS_ENCODER(ACLOwner)
 
 class RGWAccessControlPolicy
 {
 protected:
-  CephContext *cct;
   RGWAccessControlList acl;
   ACLOwner owner;
 
 public:
-  explicit RGWAccessControlPolicy(CephContext *_cct) : cct(_cct), acl(_cct) {}
-  RGWAccessControlPolicy() : cct(NULL), acl(NULL) {}
-  virtual ~RGWAccessControlPolicy() {}
```

Comment:
i think this still needs a virtual destructor because req_state stores `user_acl`, `bucket_acl`, `object_acl` as `unique_ptr<RGWAccessControlPolicy>` and allocates the s3/swift subclass depending on dialect. i'll see if i can avoid the need for virtuals altogether